### PR TITLE
Bump min supported Flutter version to 3.13.0

### PR DESCRIPTION
**Pull request by ulusoyca**
## Description

According to last 4 stable version support rule, we are bumping the min supported Flutter SDK version to 3.13.0.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ repository: https://github.com/woltapp/wolt_modal_sheet
 
 environment:
   sdk: '>=3.0.6 <4.0.0'
-  flutter: 3.10.6
+  flutter: '>=3.13.0'
```
Comment (TahaTesser) [inline pubspec.yaml:8]:
```suggestion
  flutter: 3.13.0
```

Explicit version as we currently use

Comment (TahaTesser) [reply pubspec.yaml:8]: 3.13.0 has analyzer issues in the package?

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (TahaTesser) [reply pubspec.yaml:8]: Please update our GitHub actions to use the same Flutter version.

Comment (TahaTesser) [inline pubspec.yaml:8]:
```suggestion
  # Provide explicit version for flutter-action to automatically pull this version.
  flutter: 3.13.0 
```
Might be worth documentating for future releases

Comment (TahaTesser) [reply pubspec.yaml:8]: Explicit version also required by the PR check to use Flutter version from pubspec.

Screenshot 2024-06-01 at 23 33 13

Comment (ulusoyca) [reply pubspec.yaml:8]: @TahaTesser using this as 3.13.0 instead of '>=3.13.0' gives us warnings with pub dry run.

Comment (TahaTesser) [reply pubspec.yaml:8]: @bartekpacia
You talked about `--dry-run` [here](https://github.com/subosito/flutter-action/discussions/215#discussioncomment-5382431),

if we add `>=3.13.0', it breaks our GitHub actions. Do you've any solutons?

Comment (bartekpacia) [reply pubspec.yaml:8]: Hi @TahaTesser and @ulusoyca 👋 😁

Unfortunately, you cannot use version ranges (like `>= 3.13.0`) when also using `flutter-version-file`. [The stuff I talk in comment that you mentioned](https://github.com/subosito/flutter-action/discussions/215#discussioncomment-5382431) is actually not implemented in https://github.com/subosito/flutter-action/pull/290.

Sorry about that – and PRs are welcome :)

Comment (TahaTesser) [reply pubspec.yaml:8]: @bartekpacia  Appreciate it!

@ulusoyca 
Could you please update GitHub actions with `3.13.0` and remove automatic version tag to fix our checks?

**Comment by TahaTesser** [comment]
@ulusoyca
Could you please update GitHub actions with 3.13.0 and remove automatic version tag to fix our checks?

**Comment by TahaTesser** [comment]
@ulusoyca 
Updated following deprecated APIs in `3.13.0` in https://github.com/woltapp/wolt_modal_sheet/pull/219/commits/121bad9c6ff60af4dffda3fdc25e84f214dfbbca. Please take a look if it looks good. Also fixed checks,

Screenshot 2024-06-02 at 21 51 2
2

Screenshot 2024-06-02 at 21 48 45

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -18,8 +18,8 @@ jobs:
       - uses: actions/checkout@v4
       - uses: subosito/flutter-action@v2.16.0
         with:
+          flutter-version: '3.13.0' # Uses the minimum supported version.
```
Comment (ulusoyca) [inline .github/workflows/checks.yml:21]: What is the reason we are switching back to this way?

Comment (TahaTesser) [reply .github/workflows/checks.yml:21]: Please see https://github.com/woltapp/wolt_modal_sheet/pull/219#discussion_r1623630183

Comment (TahaTesser) [reply .github/workflows/checks.yml:21]: We cannot use automatic pubspec version if we use constraints.

**Review by TahaTesser — APPROVED** [review]
I fixed checks.yaml but deploy check is still failing for some reason, not sure why.

Otherwise, LGTM!

**Comment by ulusoyca** [comment]
@AcarFurkan Can you please fix the build_and_preview workflows accordingly and push the changes to this PR? Huge thanks!

**Comment by AcarFurkan** [comment]
@ulusoyca We are probably getting this error because the existing YAML is running in main. After we get the merge we can test it with a pr.

**Pull request merged**